hls4ml/converters/onnx_to_hls.py

```diff
@@ -125,252 +183,95 @@ def onnx_to_hls(yamlConfig):
     layer_list = []
 
     #Extract model architecture
-    model = ModelProto()
-    with open(yamlConfig['OnnxModel'], 'rb') as fid:
-        model.ParseFromString(fid.read())
+    print('Interpreting Model ...')
+    if 'OnnxAPIModel' in config:
+        # Model instance passed in config from API
+        model = config['OnnxAPIModel']
+    else:
+        #Model instance passed in from "physical" file.
+        model = onnx.load(config['OnnxModel'])
     
-    #Define supported layers
-    core_operations = ['Gemm', 'BatchNormalization', 'Conv']
-    transform_operations = ['Squeeze', 'Unsqueeze', 'Transpose', 'Flatten', 'Identity', 'Reshape']
-    pool_operations = ['AveragePool', 'MaxPool']
-    merge_operations = ['Add', 'Sub', 'Mul', 'Average', 'Max', 'Min', 'Concat', 'Sum']
-    activation_operations = ['Relu', 'Tanh', 'Sigmoid', 'LeakyRelu', 'ThresholdedRelu', 'HardSigmoid', 'Elu', 'Selu', 'PRelu', 'Softmax', 'Softsign', 'Softplus']
-    supported_operations = core_operations + transform_operations + pool_operations + merge_operations + activation_operations
-
-    operation_map = {'Gemm':'Dense', 'Relu':'Activation', 'Tanh':'Activation', 'Sigmoid':'Activation',
-    'LeakyRelu':'LeakyReLU', 'ThresholdedRelu':'ThresholdedReLU', 'HardSigmoid':'Activation',
-    'Elu':'ELU', 'Selu':'Activation', 'PRelu':'PReLU', 'Softmax':'Softmax', 'Softsign':'Activation', 'Softplus':'Activation',
-    'Sum':'Add', 'Sub':'Subtract', 'Max':'Maximum', 'Min':'Minimum', 'Mul':'Multiply', 'Concat':'Concatenate'}
+    #Optimizie the model graph's before conversion
+    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
+    graph =  _hls4ml_onnx_optimizer(model.graph)
     
-    #Define layers to skip for conversion to HLS
-    skip_layers = ['Squeeze', 'Unsqueeze', 'Dropout', 'Identity', 'Flatten', 'Transpose', 'Reshape'] 
     #Map inputs of skipped layers
     inputs_map = {}
-
-    passes = ['fuse_transpose_into_gemm', 'fuse_matmul_add_bias_into_gemm', 'eliminate_nop_transpose', 'fuse_consecutive_transposes']
-    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
-    model = optimizer.optimize(model, passes)
-    model = shape_inference.infer_shapes(model) # have to infer shapes before optimizing the model
-    
     reader = ONNXDataReader(model)
+    
+    #Input layer info (assuming that there is only one input layer)
```

Comment:
Why assuming just one? You can get the list of input layers from the onnx graph and create one input layer for each as you parse.